articles/actions.md

```diff
@@ -0,0 +1,338 @@
+
+### Action Server
+
+An action server provides an action.
+Like topics and services, an action server has a name and a type.
+There is only one server per name.
+
+It is responsible for
+
+* advertising the action to other ROS entities
+* accepting or rejecting requests from one or more action clients
+* executing the action when a request is received and accepted
+* optionally providing feedback about the progress of all executing actions
+* optionally handling requests to cancel an action
+* sending the result of the action, including whether it succeed, failed, or was cancelled, to the client when the action completes
+
+### Action Client
+
+An action client requests an action to be performed and monitors its progress.
+There may be multiple clients per server; however, it is up to the server to decide how requests from multiple clients will be handled.
+
+It is responsible for
+
+* making a request to the action server
+* optionally monitoring the feedback from the action server
+* optionally requesting that the action server cancel the action execution
+* optionally checking the result of the action received from the action server
+
+## Differences between ROS 1 and ROS 2 actions
+
+### First Class Support
+In ROS 1, actions are implemented as a separate library `actionlib` which is built on top of the client libraries.
+This was done to avoid increasing the work required to create a client library in a new language, but actions turned out to be very important to packages like the navigation stack<sup>[1](#separatelib)</sup>.
+In ROS 2 actions will be included in the client library implementations.
+The work of writing a client library in a new language will be reduced by creating a common implmentation in C.
+
+### Services used for Actions
+
+In ROS 1 actions were implemented using a set of topics under a namespace taken from the action name.
+ROS 1 Services were not used because they are inherently synchronous, and actions need to be asynchronous.
+Actions also needed to send status/feedback and be cancelable.
+In ROS 2 services are asynchronous in the common C implementation, so actions will use a combination of services and topics.
+
+### Goal Identifiers
+
+In ROS 1, Action clients are responsible for creating a goal ID when submitting a goal.
+In ROS 2 the action server will be responsible for generating the goal ID and notifying the client.
+
+The server is better equiped to generate a unique goal id than the client because there may be multiple clients who could independently generate the same goal id.
+Another reason is to avoid a race condition between goal creation and cancellation that exists in ROS 1.
+In ROS 1 if a client submits a goal and immediatly tries to cancel it then the cancelation may or may not happen.
+If the cancelation is processed before the goal submission then `actionlib` will ignore the cancellation request without notifying the user's code.
```

Comment:
the term `actionlib` appears to be deprecated as you only use it twice in this document. please make this explicit, and be consistent with that you call this ROS2 feature